### ISSUE TITLE
Midpoint Review - Accessibility Feedback - BAH-MBS-SelfServ, Pre-Need Self Service, Pre-Need modernization and CaMEO Integration

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#63190]
## VFS actions
- [x] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [x] Comment on this ticket:
  - [x] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [x] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [x] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [x] Close the ticket when all feedback has been addressed.

## Thoughts/questions

-

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Must:** Echoing @erinrwhite in #63187, links that open in a new tab should have `(opens in a new tab)` or similar. When there are design constraints it's sufficient to have that text included via `aria-label` only, but I don't think that should be necessary here.
- **Must:** Super minor fix -- when you have TTY phone links in the "Need help" section beneath the form, the TTY should be in the link text rather than outside the text. When coded, the `va-telephone` component has a `tty` prop that handles formatting for both the visible link and the `aria-label`.
- **Must:** When you have the multiple responses pattern (eg. adding service dates) and the document upload pattern, the remove and delete buttons need to have some confirmation step followed by some kind of message indicating the action is complete. Basically, we want to avoid users having an accidental single-click destructive action with no indication to users that it happened besides the visual change in what's on screen. For screen reader users and screen magnification users in particular, it's easy to end up with a scenario where a button gets accidentally clicked but the user has moved on down the page and doesn't realize anything has happened. #56082 is a past instance of a team tackling this, including some videos of good behavior once fixed.
- **Must:** As mentioned in the meeting, you already asked about the [conditionally-revealed "other" field in #accessibility-help](https://dsva.slack.com/archives/C8E985R32/p1691074145032899). Definitely sync with Jeana when she's back in the office so she can provide a model for this behavior.

**Comment by ccruzgra** [comment]
@briandeconinck Thanks so much for your feedback. Our form already has the delete confirmation modal for the supporting files page, but I haven't seen this modal used for other cards like adding service dates or persons, for example, in other forms.

**Comment by ccruzgra** [comment]
@briandeconinck For the delete confirmation modal: Since this is already in place for our upload process, just not captured in the prototype, note that no other card patterns besides for uploading have a delete confirmation modal (e.g. service periods in our form or card patterns not related to uploads in other VA.gov forms). This doesn't seem to be OOB functionality. Would this need to be a future enhancement for the VA.gov Platform team that modifies OOB components?

**Comment by briandeconinck** [comment]
Hi @ccruzgra! Sorry to keep you waiting! Great that this is in place for the upload, thank you.

For the other card pattern, you're correct that it's not OOB functionality. It's (slowly) being established as a pattern in the design system and then (hopefully) built into the forms system over time. Looking back at this ticket I see I flagged it as a "must," but it should have been a "should" instead. If you come to staging with no confirmation step on the cards I will log it as an accessibility issue, but it won't be logged as a launch-blocker.

Basically: Fix it if you can, but it's going to require a little custom effort and it's totally fine if it happens later on as tech debt. Hopefully out of box functionality for cards like this over time but I don't have a timeline for when that's going to happen.
